src/Workspaces/CoreTest/ExtensionsTests/DocumentExtensionsTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.ExtensionsTests
+{
+    public class DocumentExtensionsTests
+    {
+        private readonly AdhocWorkspace _ws = new AdhocWorkspace();
+        private readonly Project _emptyProject;
+
+        public DocumentExtensionsTests()
+        {
+            _emptyProject = _ws.AddProject(
```

Comment:
Doh, you're right of course.  We really need to move the xml consumption stuff down to workspaces and unify the test workspaces.